examples/simple/Main.C

```diff
@@ -0,0 +1,148 @@
+#include "Main.decl.h"
+#include "Paratreet.h"
+
+class Main : public CBase_Main {
+  int n_treepieces; // Cannot be a readonly because of OCT decomposition
+  int cur_iteration;
+  double total_start_time;
+  double start_time;
+  bool verify;
+  paratreet::Configuration conf;
+
+  public:
+  static void initialize() {
+    BoundingBox::registerReducer();
+  }
+
+  Main(CkArgMsg* m) {
+    // mainProxy = thisProxy;
+
+    // Initialize readonly variables
+    conf.input_file = "";
+    conf.decomp_tolerance = 0.1;
+    conf.max_particles_per_tp = 1000;
+    conf.max_particles_per_leaf = 10;
+    conf.decomp_type = OCT_DECOMP;
+    conf.tree_type = OCT_TREE;
+    conf.num_iterations = 3;
+    conf.num_share_levels = 3;
+    conf.cache_share_depth= 3;
+    conf.flush_period = 1;
+
+    conf.traversalFn = [] (CProxy_TreePiece<CentroidData>& tp, int iter) {
+      tp.template startDown<GravityVisitor>();
+    };
+
+    conf.postInteractionsFn =
+      [this] (BoundingBox& universe, CProxy_TreePiece<CentroidData>& tp, int iter) {
+        if (iter == 0 && this->verify) {
+          paratreet::outputParticles(universe, tp);
+        }
+      };
+
+    verify = false;
```

Comment:
This may be potentially dangerous. If `Main` and `Driver` are not on the same PE, or `Main` is deleted... `this` will be an invalid pointer.